File: src/mobile-android/SmartSolarMicrogridAndroid/app/src/main/kotlin/com/sliit/ssmts/operator_dashboard/data/remote/dto/FinalizeTransferDto.kt
/**
 * Description: Data Transfer Objects for energy transfer finalization requests and completion receipt responses.
 */
package com.sliit.ssmts.operator_dashboard.data.remote.dto

import com.google.gson.annotations.SerializedName

/**
 * Description: Request payload dispatched to PATCH /api/reservations/{id}/finalize containing verified metered power.
 */
data class FinalizeTransferDto(
    @SerializedName("meteredEnergyKwh")
    val meteredEnergyKwh: Double,

    @SerializedName("notes")
    val notes: String? = null
)

/**
 * Description: Finalization response receipt confirming completed transaction status and committed metered kWh.
 */
data class FinalizeTransferResponseDto(
    @SerializedName("success")
    val success: Boolean,

    @SerializedName("reservationId")
    val reservationId: String,

    @SerializedName("status")
    val status: String,

    @SerializedName("meteredEnergyKwh")
    val meteredEnergyKwh: Double,

    @SerializedName("finalizedAt")
    val finalizedAt: String,

    @SerializedName("finalizedByOperator")
    val finalizedByOperator: String
)

File: src/server/SmartSolarMicrogrid.Api/DTOs/FinalizeTransferDto.cs
// Description: Data Transfer Objects for energy transfer finalization request and receipt response contracts.

namespace SmartSolarMicrogrid.Api.DTOs;

/// <summary>
/// Description: Request payload for PATCH /api/reservations/{id}/finalize with metered power reading.
/// </summary>
public class FinalizeTransferRequestDto
{
    public double MeteredEnergyKwh { get; set; }

    public string? Notes { get; set; }
}

/// <summary>
/// Description: Response payload confirming completion of energy transfer transaction.
/// </summary>
public class FinalizeTransferResponseDto
{
    public bool Success { get; set; }

    public string ReservationId { get; set; } = string.Empty;

    public string Status { get; set; } = string.Empty;

    public double MeteredEnergyKwh { get; set; }

    public DateTime FinalizedAt { get; set; }

    public string FinalizedByOperator { get; set; } = string.Empty;
}
